src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -103,20 +104,47 @@ private static string ConstructFullRuleIdentifier(ReportingDescriptor reportingD
             return fullRuleIdentifier;
         }
 
-        public static string CreateWorkItemDescription(this SarifLog log)
+        public static string CreateWorkItemDescription(this SarifLog log, Uri locationUri)
         {
-            Dictionary<string, int> resultCountsByTool = ComputeToolResultCounts(log);
-            StringBuilder templateText = new StringBuilder(@"This bug contains results for the below tool(s) and issue(s).  It was filed automatically.");
-            templateText.AppendLine();
-            templateText.AppendLine();
+            Dictionary<Run, int> resultCountsByRun = ComputeRunResultCounts(log);
+            StringBuilder templateText = new StringBuilder();
+            int runningResults = 0;
+            string toolNames = string.Empty;
+            string multipleToolsFooter = string.Empty;
+
+            Uri runRepositoryUri = resultCountsByRun.FirstOrDefault().Key?.VersionControlProvenance?.FirstOrDefault().RepositoryUri;
+            string detectionLocation = !string.IsNullOrEmpty(runRepositoryUri?.OriginalString) ? runRepositoryUri?.OriginalString : locationUri?.OriginalString;
+
+            toolNames = string.Format("'{0}'", resultCountsByRun.FirstOrDefault().Key?.Tool?.Driver?.Name);
```

Comment:
>toolNames [](start = 12, length = 9)

Since the expression `resultCountsByRun.FirstOrDefault().Key?.Tool?.Driver?.Name` can end up `null` -- or at least, your code thinks it needs to anticipate that -- you can't safely pass it to `string.Format`.

`resultCountsByRun.FirstOrDefault().Key?.Tool?.Driver?.Name ?? string.Empty`

Reply:
Good point.  Definitely several more cases of this or similar things with string.Join below.

---
In reply to: [398216527](https://github.com/microsoft/sarif-sdk/pull/1823#discussion_r398216527) [](ancestors = 398216527)